design/one-pager-stacks-security-isolation.md

```diff
@@ -0,0 +1,239 @@
+Therefore, any time a CRD is created, it will also be at the cluster scope, even if instances of the CRD are scoped to a namespace.
+This has caused confusion in the past, so it's important to clear that up here.
+
+## Allowed Resource Access
+
+As previously stated, Crossplane is not intended to be a typical/full Kubernetes cluster to run arbitrary workloads.
+Once installed into the control plane, stacks will mostly target external resources and a known set of CRDs that they own and depend on.
+Therefore, we can lock down the allowed permissions for stacks to a subset of all available permissions in the Kubernetes API.
+
+The `rbac.yaml` file that allows a stack to specify the permissions it requires will be removed and no longer supported.
+Instead, the permissions for a stack can be considered **implicit**.
+The following permissions will be granted to a stack when it is installed:
+
+* **Core Kubernetes API kinds:** Simple configuration and monitoring related types such as `ConfigMap`, `Secrets` and `Events`.
+There should be no need for a stack to create a Deployment, Pod, Volume, or other types like that.
+* **Owned CRDs:** All CRDs that the stack owns and are being installed by the stack.
+* **Depends on CRDs:** All CRDs that the stack depends on and will consume to perform its primary operations (e.g. `MySQLInstance` claim).
+
+The Stack Manager will create the necessary service account, role, and binding that includes these permissions and serves as the runtime security context that the Stack will operate in.
+This is described in greater details in the sections below.
+
+A few notes on the scope of access to CRDs:
+
+* All CRDs the stack owns and depends on are expected to be namespace scoped CRDs. Access to cluster scoped CRDs will not be granted, even for Cluster Stacks.
+* The allowed verbs for the CRDs the stack owns and depends on will be the same. Stacks will likely need full CRUD access, regardless of if the stack owns or depends on the CRD.
+
+In the future, other security modes could be enabled such as a more explicit model, and the allowed set of permissions and scope could be expanded, but we are intentionally starting now with a restrictive scope.
+We will allow feedback and real use cases from the community to inform any decisions to potentially expand this security model.
+Any future expansions to the model should be accompanied by a thoughtful design.
+
+## Cluster Stacks
+
+Cluster Stacks can be thought of as a higher level of privilege than other types of stacks, similar to how kernel modules work in Linux.
+The controllers contained in a cluster stack will usually reconcile resources that are external to the cluster, for example cloud provider managed services.
+These controllers will also likely be written as full fledged controller-runtime based implementations, but that is not a requirement.
+
+Let's look at a specific example of the GCP cluster stack to understand the security and isolation model more thoroughly.
```

Comment:
It may be worth linking to the GCP stack here if we are referring to it by name.

Reply:
I'll add a link to the GCP stack repo